es6.md

```diff
@@ -189,3 +189,35 @@ interface ImportNamespaceSpecifier {
 ```
 
 A namespace import specifier, e.g., `* as foo` in `import * as foo from "mod.js"`.
+
+## ExportDeclaration
+
+```js
+interface ExportDeclaration <: Node {
```

Comment:
Still a few, but you isolate them into their own cases (and therefore reduce the number of invalid states you have to think of when dealing with any one given node):

```
interface ExportNamedDeclaration <: Node {
  declaration: Declaration | null;
  specifiers: [ ExportSpecifier ];
  source: Literal | null;
}
```

The only invalid states are having both `declaration` and `specifiers` populated and having a non-empty `specifiers` list with a non-null `source` value. Still a few invalid states, but many fewer than in the combined scenario.

```
interface ExportDefaultDeclaration <: Node {
  declaration: Declaration | Expression;
}
```

The only invalid state possible here is allowing function or class expressions (depending on who you ask :p )

```
interface ExportBatchDeclaration <: {
  source: Literal
}
```

Zero invalid states here